search/tweets_engine.go

```diff
@@ -86,19 +86,32 @@ func (e *ESEngine) CombineResultsToSingleHit(resultsByLang map[string]*elastic.S
 		}
 
 		hit := &elastic.SearchHit{
+			Source:    (*json.RawMessage)(&source),
 			Type:      hitType,
-			Score:     &maxScore,
+			Score:     firstHit.Score,
 			InnerHits: innerHitsMap,
 		}
 
 		result.Hits.Hits = []*elastic.SearchHit{hit}
 		result.Hits.TotalHits = 1
-		result.Hits.MaxScore = &maxScore
+		result.Hits.MaxScore = firstHit.Score
 	}
 
 	return resultsByLang, nil
 }
 
+func findFirstHitWithEffectiveDate(hits []*elastic.SearchHit) (*elastic.SearchHit, error) {
```

Comment:
@YuriHechter for me it's all time return error (EffectiveDate for all hits are nil)

Reply:
even if you sort by time?

Reply:
yes, for me it's nil all time